mpcpy/exodata.py

```diff
@@ -1332,7 +1337,101 @@ def _collect_data(self, start_time, final_time):
+    weaForeModel: Weather forecast model, str,
+        GFS: Global Forecast System model, available for the entire globe and for 7 days ahead, support historical data, updated every 6 hours, 
+             time resolution: 3 hours, geographical resolutions: 0.25 and 0.5 deg 
+        HRRR: High Resolution Rapid Refresh model, available US and for ~15 hours ahead, DOES NOT support historical data, updated every hour, 
+             time resolution: 1 hour, geographical resolutions: 3 km
+        RAP: Rapid Refresh model, available US and for 2 days ahead (tested 18 hours), support historical data, updated every hour, 
+             time resolution: 1 hour, geographical resolutions: 20, 40 km
+        NAM: North American Mesoscale model, available for the whole North America and for 4 days ahead (tested 3 days), support historical data, 
+             updated every 6 hours, time resolution: 1 hour, geographical resolutions: 20 km
+    tz_name: timezone name, str
+        Default is 'from geography', then the timezone would be inferred automatically from the input geography
+        Users could also specify the timezone manually, example 'America/Chicago', 'UTC'
+
+    Attributes
+    ----------
+    data : dictionary
+        {"Weather Variable Name" : mpcpy.Variables.Timeseries}.
+    lat : mpcpy.variables.Static
+        Latitude in degrees.
+    lon : mpcpy.variables.Static
+        Longitude in degrees.
+    tz_name : string
+        Timezone name.  
+
+    '''
+    
+    def __init__(self, geography, weaForeModel, tz_name='from_geography', **kwargs):
+        '''Constructor of DataFrame weather exodata object.
+        
+        '''
+        
+        self.name = 'weather_from_noaa';
+        self.data = {};   
+        # Dictionary of format {'dfHeader' : ('weaVarName', mpcpyUnit)}
+        self.variable_map = {'temp_air'     : ('weaTDryBul', units.degC),
+                             'wind_speed'   : ('weaWinSpe', units.m_s),
+                             'ghi'          : ('weaHGloHor', units.W_m2),
+                             'dni'          : ('weaHDirNor', units.W_m2),
+                             'dhi'          : ('weaHDifHor', units.W_m2),
+                             'total_clouds' : ('weaNTot', units.percent),
+                             };
+        self.geography = geography
+        self.lat = variables.Static('lat', self.geography[0], units.deg);
+        self.lon = variables.Static('lon', self.geography[1], units.deg)
+        
+        # Initialize the weather forecast model
+        if weaForeModel == 'GFS':
+            self.model = GFS()
+        elif weaForeModel == 'HRRR':
+            self.model = HRRR()
+        elif weaForeModel == 'RAP':
+            self.model = RAP()
+        elif weaForeModel == 'NAM':
+            self.model = NAM()
+        else:
+            raise NameError('The {} forecast model is not supported. Only GFS, HRRR, RAP, NAM are supported now'.format(method))
```

Comment:
I think ``method`` should be replaced with ``weaForeModel``.  In any case, I think just ``model`` or ``method`` is a better parameter name than ``weaForeModel``.

Please add a unittest that tests catching this error.  You can look at usage of ``assertRaises`` elsewhere in unittests for guidance.